lightning/src/ln/onion_route_tests.rs

```diff
@@ -1397,9 +1397,12 @@ fn test_phantom_failure_modified_cltv() {
 	commitment_signed_dance!(nodes[0], nodes[1], update_1.commitment_signed, false);
 
 	// Ensure the payment fails with the expected error.
+	let mut err_data = Vec::new();
+	err_data.extend_from_slice(&update_add.cltv_expiry.to_be_bytes());
+	err_data.extend_from_slice(&0u16.to_be_bytes());
 	let mut fail_conditions = PaymentFailedConditions::new()
 		.blamed_scid(phantom_scid)
-		.expected_htlc_error_data(0x2000 | 2, &[]);
+		.expected_htlc_error_data(0x1000 | 13, &err_data);
```

Comment:
Mh may be a stupid question because maybe I did not look enough inside the test code. However, why do we need to modify this part of the test?

Reply:
In the case where we forward to phantom channels, we previously overrode the `0x1000` error code with an `0x2000 | 2` error code, as we did not have the appropriate `channel_update` for that channel.

This `channel_update` is no longer required, so we are able to get rid of this override, and return the expected error code.

See the snippet below from further above:
```
} else if err_code & 0x1000 == 0x1000 {
			// If we're trying to return an error that requires a `channel_update` but
			// we're forwarding to a phantom or intercept "channel" (i.e. cannot
			// generate an update), just use the generic "temporary_node_failure"
			// instead.
			err_code = 0x2000 | 2;
```